object_db.go

```diff
@@ -383,3 +376,14 @@ func (o *ObjectDatabase) cleanup(f *os.File) {
 	f.Close()
 	os.Remove(f.Name())
 }
+
+func hasher(algo string) hash.Hash {
```

Comment:
I wouldn't be upset if this patch came first (since the introduction of this code in `Hasher()` and subsequent move here creates a little more reviewer burden, but I think that you could just as easily ignore this.

Reply:
I've just squashed it into the appropriate patch earlier in the series.